include/jwt-cpp/jwt.h

```diff
@@ -2606,12 +2607,166 @@ namespace jwt {
 #endif
 	};
 
+	namespace verify_ops {
+		template<typename json_traits>
+		struct verify_context {
+			verify_context(date ctime, const decoded_jwt<json_traits>& j, size_t l)
+				: current_time(ctime), jwt(j), default_leeway(l) {}
+			// Current time, retrieved from the verifiers clock and cached for performance and consistency
+			date current_time;
+			// The jwt passed to the verifier
+			const decoded_jwt<json_traits>& jwt;
+			// The configured default leeway for this verification
+			size_t default_leeway{0};
+
+			// The claim key to apply this comparision on
+			typename json_traits::string_type claim_key{};
+		};
+
+		/**
+		 * This is the default operation and does case sensitive matching
+		 */
+		template<typename json_traits>
+		struct equals_claim {
+			basic_claim<json_traits> expected;
+			void operator()(const verify_context<json_traits>& ctx, std::error_code& ec) {
+				if (!ctx.jwt.has_payload_claim(ctx.claim_key)) {
+					ec = error::token_verification_error::missing_claim;
+					return;
+				}
+				auto jc = ctx.jwt.get_payload_claim(ctx.claim_key);
+				if (jc.get_type() != expected.get_type()) {
+					ec = error::token_verification_error::claim_type_missmatch;
+					return;
+				}
+				bool matches = false;
+				switch (expected.get_type()) {
+				case json::type::boolean: matches = expected.as_bool() == jc.as_bool(); break;
+				case json::type::integer: matches = expected.as_int() == jc.as_int(); break;
+				case json::type::number: matches = expected.as_number() == jc.as_number(); break;
+				case json::type::string: matches = expected.as_string() == jc.as_string(); break;
+				case json::type::array:
+				case json::type::object:
+					matches = json_traits::serialize(expected.to_json()) == json_traits::serialize(jc.to_json());
```

Comment:
💭 I wonder if we can let the underlying JSON library handle this.. might be a fun test

Reply:
Might be a good idea, since most probably have some sort of way to compare json values. However it would be an api breaking change, since we'd need to extend the json traits (and everyone using custom json libraries would need to do the same).

Reply:
Thinking about it we might be able to detect if the "equals" method is part of the json_traits with some template magic and switch between the implementations we have now and the traits method using specialization or enable_if, without breaking everyones code. But we should probably change that everywhere we compare json values then.